controllers/bucket_controller.go

```diff
@@ -188,3 +208,61 @@ func removeKey(slice []string, s string) []string {
 	}
 	return slice
 }
+
+func (b *BucketReconciler) WaitForSecret(namespace, name string) (*corev1.Secret, error) {
+	// set a timeout of 10 minutes
+	timeout := time.After(10 * time.Minute)
+
+	// set a polling interval of 10 seconds
+	ticker := time.NewTicker(10 * time.Second)
+
+	// loop until the timeout or the secret is found
+	for {
+		select {
+		case <-timeout:
+			// timeout is exceeded, return an error
+			return nil, fmt.Errorf("timed out waiting for secret %s in namespace %s", name, namespace)
+			// add to this error with a pointer to instructions for following a manual path to a Secret that will work on STS
+		case <-ticker.C:
+			// polling interval is reached, try to get the secret
+			key := types.NamespacedName{
+				Name:      name,
+				Namespace: namespace,
+			}
+			secret := corev1.Secret{}
+			err := b.Client.Get(context.Background(), key, &secret)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					// secret does not exist yet, continue waiting
+					continue
+				} else {
+					// some other error occurred, return it
+					return nil, err
+				}
+			} else {
+				// secret is found, return it
+				return &secret, nil
+			}
+		}
+	}
+}
+
+func (b *BucketReconciler) SharedCredentialsFileFromSecret(secret *corev1.Secret) (string, error) {
+	var data []byte
+	switch {
```

Comment:
nit: if len is <= 0 error, and then use secret.Data["credentials"] instead of using a new byte array. I guess the bigger nit is I would not use a switch statement for this but this is a nit, the logic part is sound.

Reply:
Done